src/Test/nlp.jl

```diff
@@ -337,3 +337,142 @@ const nlptests = Dict("hs071" => hs071_test,
                       )
 
 @moitestset nlp
+
+"""
+    test_linear_mixed_complementarity(model::MOI.ModelLike, config::TestConfig)
+
+Test the solution of the linear mixed-complementarity problem:
+`F(x) complements x`, where `F(x) = M * x .+ q` and `0 <= x <= 10`.
+"""
+function test_linear_mixed_complementarity(model::MOI.ModelLike, config::TestConfig)
+    MOI.empty!(model)
+    x = MOI.add_variables(model, 4)
+    MOI.add_constraint.(model, MOI.SingleVariable.(x), MOI.Interval(0.0, 10.0))
+    MOI.set.(model, MOI.VariablePrimalStart(), x, 0.0)
+    M = Float64[0  0 -1 -1; 0  0  1 -2; 1 -1  2 -2; 1  2 -2  4]
+    q = [2; 2; -2; -6]
+    terms = MOI.VectorAffineTerm{Float64}[]
+    for i = 1:4
+        push!(
+            terms,
+            MOI.VectorAffineTerm(4 + i, MOI.ScalarAffineTerm(1.0, x[i]))
+        )
+        for j = 1:4
+            iszero(M[i, j]) && continue
+            push!(
+                terms,
+                MOI.VectorAffineTerm(1, MOI.ScalarAffineTerm(M[i, j], x[j]))
```

Comment:
I have some trouble figuring out why we set `1` there...

Reply:
Oops. That should be `i`.

Reply:
Ok, that's make sense!